engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/AtomConstructor.java

```diff
@@ -258,6 +259,20 @@ Atom instantiate(Object... arguments) throws ArityException {
     return newInstance(arguments);
   }
 
+  @ExportMessage
+  TriState isIdenticalOrUndefined(Object other) {
+    if (other instanceof AtomConstructor otherCtor) {
+      return this == otherCtor ? TriState.TRUE : TriState.FALSE;
+    } else {
+      return TriState.FALSE;
+    }
+  }
+
+  @ExportMessage
+  int identityHashCode() {
+    return System.identityHashCode(this);
```

Comment:
This will make the hash code random in different runs. That's fine according to Java `hashCode` behavior, but we might have it more stable knowing `AtomConstructor` have well defined name.

`java.lang.reflect.Method` also tries to assign more stable `hashCode`, btw. Surprisingly `java.lang.Class` doesn't - probably for compatibility reasons as they forget to do it in Java 1.0...

Reply:
I got rid of all `identityHashCode` messages from all our classes. They were a legacy from my very first prototype implementation. All the functionality is now implemented in `HashCodeAnyNode`.